### PR TITLE
Add qml.CY operation

### PR DESCRIPTION
**Pull request by josh146**
**Context:** PennyLane currently has the CX (CNOT) and CZ gates, but does not provide the CY operation.

**Description of the Change:**

* Adds the CY operation to PennyLane, as `qml.CY`.

* Adds the decomposition `CY = S(0) @ CRY(pi/2, [0, 1])` for devices that don't support the CY gate natively. If a device does not support CRY, this will also be decomposed automatically.

* Adds tests.

**Benefits:** n/a

**Possible Drawbacks:** n/a

**Related GitHub Issues:** n/a

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -7,53 +7,59 @@
...
+
+
 <h3>Improvements</h3>
 
-* Sped up the application of certain gates in ``default.qubit`` by using array/tensor
-  manipulation tricks. The following gates are affected: ``PauliX``, ``PauliY``, ``PauliZ``,
-  ``Hadamard``, ``SWAP``, ``S``, ``T``, ``CNOT``, ``CZ``.
+* Sped up the application of certain gates in `default.qubit` by using array/tensor
+  manipulation tricks. The following gates are affected: `PauliX`, `PauliY`, `PauliZ`,
+  `Hadamard`, `SWAP`, `S`, `T`, `CNOT`, `CZ`.
```
Comment (josh146) [inline .github/CHANGELOG.md:27]: My text editor made some unsanctioned (but correct) changes!

**Comment by josh146** [comment]
Not sure why codecov is complaining, coverage seems okay to me 🤔 
image

File: tests/ops/test_qubit_ops.py [hunk 2]
```diff
@@ -461,6 +461,21 @@ def test_phase_decomposition(self, tol):
...
+
+        mats = []
+        for i in reversed(res):
+            if len(i.wires) == 1:
+                mats.append(np.kron(i.matrix, np.eye(2)))
+            else:
+                mats.append(i.matrix)
+
+        decomposed_matrix = np.linalg.multi_dot(mats)
+        assert np.allclose(decomposed_matrix, op.matrix, atol=tol, rtol=0)
```
Comment (thisac) [inline tests/ops/test_qubit_ops.py:477]: Nice, simple test. :tada:

File: pennylane/ops/qubit.py [hunk 3]
```diff
@@ -364,6 +364,49 @@ def _eigvals(cls, *params):
...
+            [1, 0, 0, 0],
+            [0, 1, 0, 0],
+            [0, 0, 0, -1j],
+            [0, 0, 1j, 0],
+        ]
+    )
+
+    @classmethod
+    def _matrix(cls, *params):
+        return cls.matrix
```
Comment (thisac) [inline pennylane/ops/qubit.py:402]: Is this the "main implementation" of a gate? In other words, is this method the one that is used for all calculations when a particular gate is in circuits?

**Review by thisac — APPROVED** [review]
Nice! :1st_place_medal:

File: pennylane/ops/qubit.py (hunk 3 above)
Comment (josh146) [reply pennylane/ops/qubit.py:402]: It's up to the plugin. Some plugins may choose to read the matrix property, and use this to implement the gate (e.g., `default.qubit`). Others may have their own implementation, that doesn't require accessing this property.

**Pull request merged**